### ISSUE TITLE
Charts Not Built for arm64, armv7 targets in Xcode 12 beta 6

### ISSUE THREAD

**Issue by sandesh-gauss** [danielgindi/Charts#4468]
Hi,

I am having issues building the Charts framework with the Xcode 12 Beta 6. Getting the following error,

Could not find module 'Charts' for target 'arm64-apple-ios-simulator'; found: x86_64-apple-ios-simulator, x86_64

Works fine with Xcode 12 initial beta. Please let me know if there's a quck fix for the time being.

**Comment by quifago** [comment]
If you were using Carthage, you can use the script they mention in their Issues page:
https://github.com/Carthage/Carthage/issues/3019

It worked for me.

**Comment by bobek-balinek** [comment]
I have had the same issue when bundling Charts through Swift Package Manager. My workaround for Debug configuration:

- Build Active Architectures Only (`ONLY_ACTIVE_ARCH`): `YES`
- Valid Architectures (`VALID_ARCHS`): `arm64 arm64e armv7 armv7s i386 x86_64`

The last one was set as a user-defined variable, no idea why and how it's worked ¯\_(ツ)_/¯

**Comment by quifago** [comment]
@bobek-balinek by any chance, have you tried to checkout the v4.0.0 branch? When I tried it few weeks ago it was not working, not sure if your workaround could fix it too. And that's the version of Charts that we're using in our app.

**Comment by sandesh-gauss** [comment]
> I have had the same issue when bundling Charts through Swift Package Manager. My workaround for Debug configuration:
> 
> * Build Active Architectures Only (`ONLY_ACTIVE_ARCH`): `YES`
> * Valid Architectures (`VALID_ARCHS`): `arm64 arm64e armv7 armv7s i386 x86_64`
> 
> The last one was set as a user-defined variable, no idea why and how it's worked ¯_(ツ)_/¯

@bobek-balinek  Yes, thank you fo your comment. I am using Swift Package Manager as well. I tried that before, but the thing is the iOS app I am using has some Camera APIs that only work with arm64 arm64e armv7 armv7s. This is definitely a great workaround if adding those i386 x86_64 doesn't break other code in your app. I think the solution is as simple as re-compiling the Charts framework with the arm64 and above architectures.

**Comment by Seoras** [comment]
With iOS14 release today I update to Xcode 12 via the Mac App Store and now my project which uses Charts fails to build with the following errors:

> <unknown>:0: error: module map file '/Users/george/Xcode/DerivedData/Xcode DrvdData/.../Build/Products/Debug-iphonesimulator/Charts/Charts.modulemap' not found
> <unknown>:0: error: module map file '/Users/george/Xcode/DerivedData/Xcode DrvdData/.../Build/Products/Debug-iphonesimulator/Charts/Charts.modulemap' not found
> <unknown>:0: error: missing required module 'SwiftShims'

The project is primarily ObjC. A few years ago I added a dummy Swift file to prompt Xcode to make a Bridging Header and all has worked well like this until today and Xcode 12.

Looking in the DerivedData/Build/Products/Debug-iphonesimulator/ folder I see no files, other than the _app-name_.app file.
I suspect others may not be reporting this yet since they may not have executed a "make clean" or cleaned out their DerivedData folder which I did before launching Xcode 12 for the first time.

I'm assuming this issue is the same/related?
